internal/integration/provision/upgrade.go

```diff
@@ -136,6 +136,29 @@ func upgradeStableReleaseToCurrent() upgradeSpec {
 	}
 }
 
+// upgradeCurrentReleaseToCurrent upgrades current version to the current version of Talos.
+func upgradeCurrentReleaseToCurrent() upgradeSpec {
+	installerImage := fmt.Sprintf("%s/%s:%s", DefaultSettings.TargetInstallImageRegistry, images.DefaultInstallerImageName, DefaultSettings.CurrentVersion)
+	return upgradeSpec{
+		ShortName: fmt.Sprintf("%s-%s", stableRelease, DefaultSettings.CurrentVersion),
+
+		SourceKernelPath:     helpers.ArtifactPath(filepath.Join(trimVersion(stableRelease), constants.KernelAsset)),
```

Comment:
this is not technically correct, as we're booting previous release, but we can leave it for now until we see a problem. for now 0.8 should be able to install 0.9

Reply:
well, actually it was causing issues. It can install 0.9, but not in the encrypted mode, because old installer formats the partition ;)